File: PA_33.py
#LC 75 - Sort Colors
#Time Complexity - O(n)
#Space Complexity - O(1)
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        if not nums:
            return []
        
        low = 0
        mid = 0
        high = len(nums)-1
        
        while mid <= high:
            if nums[mid] == 1:
                mid = mid + 1
            elif nums[mid] == 2:
                temp = nums[high]
                nums[high] = nums[mid]
                nums[mid] = temp
                high = high - 1
            elif nums[mid] == 0:
                temp = nums[low]
                nums[low] = nums[mid]
                nums[mid] = temp
                low = low + 1
                mid = mid + 1
        #print(nums)

File: PA33.java
//Leetcode 75. Sort Colors
//Time complexity O(n)
//Space Complexity O(1)

class Solution {
    public void sortColors(int[] nums) {
        int l=0;
        int m=0;
        if(nums==null || nums.length==0 )
        {
            return;
        }
        int h= nums.length-1;
        while(m<=h){
            if(nums[m]==2){
                int temp = nums[h];
                nums[h]= nums[m];
                nums[m]= temp;
                h--;
            }else{
                if(nums[m]==1){
                    m++;
                }
            
            else{
                int temp = nums[l];
                nums[l]= nums[m];
                nums[m]= temp;
                m++;
                l++;
            }
        }
    }
 
    }
}